### PR TITLE
fix: distribute iterator value invalid

### LINKED ISSUE #2931: _*_cate output different result
**Bug Description**
while partition setting = 2,
*_cate may get different answer

**Expected Behavior**
one true result

**Relation Case**
CREATE TABLE mt ( col_s0 string,col_s1 string,col_s2 string,col_s3 string,col_s4 string,col_s5 string,col_s6 string,col_s7 string,col_s8 string,col_s9 string,col_s10 string,col_s11 string,col_s12 string,col_s13 string,col_s14 string,col_d0 double,col_d1 double,col_d2 double,col_d3 double,col_d4 double,col_t0 timestamp,col_t1 timestamp,col_t2 timestamp,col_t3 timestamp,col_t4 timestamp,col_i0 bigint,col_i1 bigint,col_i2 bigint,col_i3 bigint,col_i4 bigint,index(key = col_s0, ttl=0m, ttl_type=absolute, ts = col_t0),index(key = col_s1, ttl=0m, ttl_type=absolute, ts = col_t0),) OPTIONS (REPLICANUM = 1,PARTITIONNUM=2);

insert into mt values('4','val1','val1','val1','val1','val1','val1','val1','val1','val1','val1','val1','val1','val1000000','val1000000',1.500000,1.400000,1.400000,1.400000,1.400000,1667130000000,1652321552000,1652321552000,1652321552000,1652321552000,25,26,27,28,29);
insert into mt values('3','val2','val1','val1','val1','val1','val1','val1','val1','val1','val1','val1','val1','val1000000','val1000000',1.500000,1.400000,1.400000,1.400000,1.400000,1667130000000,1652321552000,1652321552000,1652321552000,1652321552000,25,26,27,28,29);
insert into mt values('3','val2','val1','val1','val1','val1','val1','val1','val1','val1','val1','val1','val1','val1000000','val1000000',1.400000,1.400000,1.400000,1.400000,1.400000,1667130000000,1652321552000,1652321552000,1652321552000,1652321552000,25,26,27,28,29);
insert into mt values('3','val1','val1','val1','val1','val1','val1','val1','val1','val1','val1','val1','val1','val1000000','val1000000',1.400000,1.400000,1.400000,1.400000,1.400000,1667130000000,1652321552000,1652321552000,1652321552000,1652321552000,25,26,27,28,29);
insert into mt values('5','val2','val1','val1','val1','val1','val1','val1','val1','val1','val1','val1','val1','val1000000','val1000000',1.600000,1.400000,1.400000,1.400000,1.400000,1667130000000,1652321552000,1652321552000,1652321552000,1652321552000,25,26,27,28,29);
**Steps to Reproduce**

1. SELECT count_cate(col_d0,col_s1) from mt;
 ----------------------------
  count_cate(col_d0, col_s1)
 ----------------------------
  :1,val2:2,val2:2
 ----------------------------
 ----------------------------
  count_cate(col_d0, col_s1)
 ----------------------------
  val1:2,val2:3
 ----------------------------

2.  SELECT max_cate(col_d0,col_s1) from mt;
 -----------------------------
  max_cate(col_d0, col_s1)
 -----------------------------
  val1:1.500000,val2:1.600000
 -----------------------------
 -------------------------------------------
  max_cate(col_d0, col_s1)
 -------------------------------------------
  :1.500000,val2:1.500000,val2:1.600000
 -------------------------------------------

### PR DESCRIPTION
**Pull request by zhanghaohit**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Fixes #2931

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

### CODE REVIEW

**Comment by zhanghaohit** [comment]
It seems that we do not copy the `Row` and maintain in some places. Only get the pointer and size.

```
  %11 = call i8* @next.iterator_row(%fe.iterator_ref_row* %iter_ref_alloca)
  %12 = call i8* @hybridse_storage_get_row_slice(i8* %11, i64 0)
  %13 = call i64 @hybridse_storage_get_row_slice_size(i8* %11, i64 0)
```

File: src/catalog/distribute_iterator.h [hunk 1]
```diff
@@ -124,9 +125,9 @@ class RemoteWindowIterator : public ::hybridse::vm::RowIterator {
     uint32_t pid_;
     std::string index_name_;
     std::shared_ptr<::openmldb::base::KvIterator> kv_it_;
-    std::vector<std::shared_ptr<::google::protobuf::Message>> response_vec_;
```
Comment (zhanghaohit) [inline src/catalog/distribute_iterator.h:127]: `response_vec_` seems not necessary as we `malloc` a new slice for every value.

File: src/catalog/distribute_iterator.cc [hunk 2]
```diff
@@ -422,6 +417,7 @@ const ::hybridse::codec::Row& RemoteWindowIterator::GetValue() {
     int8_t* copyed_row_data = reinterpret_cast<int8_t*>(malloc(sz));
     memcpy(copyed_row_data, slice_row.data(), sz);
     auto shared_slice = ::hybridse::base::RefCountedSlice::CreateManaged(copyed_row_data, sz);
+    buffered_slices_.push_back(shared_slice);
```
Comment (zhanghaohit) [inline src/catalog/distribute_iterator.cc]: batch sliding window no need

File: src/catalog/distribute_iterator.h [hunk 3]
```diff
@@ -124,9 +124,9 @@ class RemoteWindowIterator : public ::hybridse::vm::RowIterator {
...
     std::shared_ptr<openmldb::client::TabletClient> tablet_client_;
     ::hybridse::codec::Row row_;
+    std::vector<hybridse::base::RefCountedSlice> buffered_slices_;
```
Comment (dl239) [inline src/catalog/distribute_iterator.h]: rm unused `buffered_slices_` ?

**Review by dl239 — APPROVED** [review]

**Comment by zhanghaohit** [comment]
long-term solution is recorded in #2944

**Review by aceforeverd — APPROVED** [review]

**Pull request merged**